eva-tools/src/main/java/embl/ebi/variation/eva/vcfDump/VariantExporter.java

```diff
@@ -0,0 +1,350 @@
+        }
+//        return VariantFileMetadataToVCFHeaderConverter.parseVcfHeader(fileHeader);
+
+        // */
+
+    }
+
+    public int getFailedVariants() {
+        return failedVariants;
+    }
+
+    class VariantFields {
+        public int start;
+        public int end;
+        public String reference;
+        public String alternate;
+    }
+
+    /**
+     * converts org.opencb.biodata.models.variant.Variant into one or more htsjdk.variant.variantcontext.VariantContext
+     * behaviour:
+     * * one VariantContext per study
+     * * split multiallelic variant will remain split.
+     * * in case a normalized INDEL has empty alleles, the original alleles in the vcf line will be used.
+     *
+     * steps:
+     * * foreach variantSourceEntry, collect genotypes in its study, only if the study was requested
+     * * get main variant data: position, alleles, filter...
+     * * if there are empty alleles, get them from the vcf line
+     * * get the genotypes
+     * * add all (position, alleles, genotypes...) to a VariantContext for each study.
+     *
+     * @param variant
+     * @return
+     */
+    public Map<String, VariantContext> convertBiodataVariantToVariantContext(
+            Variant variant, Map<String, VariantSource> sources) 
+            throws IOException {
+
+        List<String> studyIds = new ArrayList<>(sources.keySet());
+        Map<String, VariantContext> variantContextMap = new TreeMap<>();
+        VariantContextBuilder variantContextBuilder = new VariantContextBuilder();
+
+        Integer start = variant.getStart();
+        Integer end = variant.getEnd();
+        String reference = variant.getReference();
+        String alternate = variant.getAlternate();
+        String filter = "PASS";
+        List<String> allelesArray = Arrays.asList(reference, alternate);
+        ArrayList<Genotype> genotypes = new ArrayList<>();
+        Map<String, List<Genotype>> genotypesPerStudy = new TreeMap<>();
+        
+        for (Map.Entry<String, VariantSourceEntry> source : variant.getSourceEntries().entrySet()) {
+
+            String studyId = source.getValue().getStudyId();
+
+            if (studyIds.contains(studyId)) {   // skipping studies not asked
+                
+                // if we added this outside the loop, if the study is not present in this variant, the writer would add 
+                // a whole line of "./."
+                if (!genotypesPerStudy.containsKey(studyId)) {  
+                    genotypesPerStudy.put(studyId, new ArrayList<Genotype>());
+                }
+                
+                // if there are indels, we cannot use the normalized alleles, (hts forbids empty alleles) so we have to take them from the original vcf line
+                int emtpyAlleles = 0;
+                for (String a : allelesArray) {
+                    emtpyAlleles += a.isEmpty() ? 1 : 0;
```

Comment:
It doesn't really matter how many empty alleles are there. As soon as one is found, the loop could be broken.